pkg/registry/apis/provisioning/repository/go-git/wrapper.go

```diff
@@ -351,39 +354,45 @@ func (g *GoGitRepo) Write(ctx context.Context, fpath string, ref string, data []
 			When:  sig.When,
 		}
 	}
-	_, err = g.tree.Commit(message, opts)
+	_, err := g.tree.Commit(message, opts)
 	if errors.Is(err, git.ErrEmptyCommit) {
 		return nil // empty commit is fine -- no change
 	}
 	return err
 }
 
 // Delete implements repository.Repository.
-func (g *GoGitRepo) Delete(ctx context.Context, path string, ref string, message string) error {
-	if _, err := g.tree.Remove(safepath.Join(g.config.Spec.GitHub.Path, path)); err != nil {
+func (g *GoGitRepo) Delete(ctx context.Context, fpath string, ref string, message string) error {
+	fpath = safepath.Join(g.config.Spec.GitHub.Path, fpath)
+	if err := verifyPathWithoutRef(fpath, ref); err != nil {
 		return err
 	}
-
-	return nil
+	if _, err := g.tree.Remove(fpath); err != nil {
+		return err
+	}
+	return g.maybeCommit(ctx, message)
 }
 
 // Read implements repository.Repository.
-func (g *GoGitRepo) Read(ctx context.Context, path string, ref string) (*repository.FileInfo, error) {
-	readPath := safepath.Join(g.config.Spec.GitHub.Path, path)
-	stat, err := g.tree.Filesystem.Lstat(readPath)
+func (g *GoGitRepo) Read(ctx context.Context, fpath string, ref string) (*repository.FileInfo, error) {
+	fpath = safepath.Join(g.config.Spec.GitHub.Path, fpath)
```

Comment:
```suggestion
	fpath = safepath.Join(g.config.Spec.GitHub.Path, fpath)
```


We receive / return relative paths through the repository interface. So let's revert this change. 

Not sure what triggered this change. Maybe you can explain.

Reply:
good catch -- thank you!  we need to add the `verifyPathWithoutRef()` check since ref is not supported in GoGitRepo yet!

I copied all the code from elsewhere along with the paths 💣